### PR TITLE
Components: Add a general promo card component

### LINKED ISSUE #34529: Earn Section: Create a CTA card component
## Description

This can be modeled after the existing [marketing section CTA card](https://github.com/Automattic/wp-calypso/pull/32510/files#diff-a9a814f97b424ca36613aca77c1f7bc0R21).

The goal will be to abstract this out into a generic component that can then be reused within multiple sections.

## Related

- https://github.com/Automattic/wp-calypso/issues/34530
- https://github.com/Automattic/wp-calypso/issues/34528

### PR DESCRIPTION
**Pull request by pablinos**
As part of the project to overhaul the 'Earn' section, we need a card component that is similar to the card used in the Marketing Tools section.

This change creates a generalised version of the card component that can be used in both sections, along with any future sections that follow a similar pattern.

#### Testing instructions

The new component can be seen in the Devdocs at `/devdocs/design/promo-card` This tests uses the version that tests for the memberships/recurring payments feature and displays the button encouraging an upgrade.

Closes #34529

### CODE REVIEW

**Comment by pablinos** [comment]
Nice! Thanks for looking at this @sirreal. I'll tidy things up. It's still work in progress but wanted to get some feedback.

File: client/components/promo-section/promo-card/cta.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import Button from 'components/button';
+
+interface CtaButton {
+	isPrimary: boolean;
+	text: string;
+	url: string;
+}
+
+interface Props {
+	button: CtaButton;
```
[the remark this answers is not in the document]
Comment (pablinos) [reply client/components/promo-section/promo-card/cta.tsx]: My nesting has got worse in the current iteration, sorry! Any suggestions for improvement are very welcome.

File: client/components/promo-section/promo-card/cta.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+		) : (
+			<Button { ...ctaBtnProps( cta.upgradeButton ) }>{ cta.upgradeButton.text }</Button>
+		);
+	}
+	return (
+		<ActionPanelCta>
+			{ ctaBtn }
+			{ learnMoreLink && (
+				<Button borderless className="promo-card__cta-learn-more" href="{ learnMoreLink }">
+					Learn More
```
Comment (eliorivero) [inline client/components/promo-section/promo-card/cta.tsx]: This should be translatable

Comment (eliorivero) [reply client/components/promo-section/promo-card/cta.tsx]: Pushed a commit to solve this

**Comment by eliorivero** [comment]
In 6857e93
Make "Learn more" translatable.
Remove double quotes around braces in `<Button borderless className="promo-card__cta-learn-more" href="{ learnMoreLink }">` so `learnMoreLink` value is used. Otherwise it was being used as a string.

**Pull request merged**